ladybug/windrose.py

```diff
@@ -0,0 +1,953 @@
+            self._show_zeros = True
+        return self._show_zeros
+
+    @show_zeros.setter
+    def show_zeros(self, show_zeros):
+        """Set the show_zeros boolean.
+
+        Args:
+            show_zeros: Boolean
+        """
+        self._show_zeros = show_zeros
+
+    @property
+    def frequency_spacing_distance(self):
+        """Distance for frequency lines.
+
+        Default is 0.1
+        """
+        if self._frequency_spacing_distance is None:
+            self._frequency_spacing_distance = 0.1
+        return self._frequency_spacing_distance
+
+    @frequency_spacing_distance.setter
+    def frequency_spacing_distance(self, frequency_spacing_distance):
+        """Set the frequency spacing distance for the windrose plot.
+
+        Args:
+            frequency_spacing_distance: Float representing frequency gap distance.
+        """
+        self._frequency_spacing_distance = frequency_spacing_distance
+
+    @property
+    def frequency_maximum(self):
+        """Maximum frequency to represent in the windrose plot.
+
+        Default will be the maximum number of items in current windrose data.
+        """
+        if self._frequency_maximum is None:
+            self._frequency_maximum = self.real_freq_max
+        return self._frequency_maximum
+
+    @frequency_maximum.setter
+    def frequency_maximum(self, frequency_maximum):
+        """Set the frequency maximum for the windrose plot.
+
+        Args:
+            frequency_maximum: Float representing frequency maximum.
+        """
+        self._frequency_maximum = frequency_maximum
+
+    @property
+    def label_point_size_factor(self):
+        """Get the size of the points for labels in the windrose plot.
+
+        Default is 0.8
+        """
+        if self._label_point_size_factor is None:
+            self._label_point_size_factor = 0.8
+        return self._label_point_size_factor
+
+    @label_point_size_factor.setter
+    def label_point_size_factor(self, label_point_size_factor):
+        """Set the size of the points for labels in the windrose plot."""
+        self._label_point_size_factor = label_point_size_factor
+
+    @property
+    def label_tick_size_factor(self):
+        """Get the size of the ticks for labels in the windrose plot.
+
+        Default is 0.3
+        """
+        if self._label_tick_size_factor is None:
+            self._label_tick_size_factor = 0.3
+        return self._label_tick_size_factor
+
+    @label_tick_size_factor.setter
+    def label_tick_size_factor(self, label_tick_size_factor):
+        """Set the size of the ticks for labels in the windrose plot."""
+        self._label_tick_size_factor = label_tick_size_factor
+
+    @property
+    def legend_parameters(self):
+        """The LegendParameters object for this plot.
+
+        Default is a legend parameter with minimum and maximum values of the
+        windrose analysis values. If show_stack is False, then these min and max
```

Comment:
I just realised I need to rewrite the histogram_data to show a nested, nested data structure (list of 'stack-intervals' nested within list of bins) to accurately show the show_stack values, since it's the average of the stack-intervals that represent the range of the windrose when show-stack = True, not the real min/max values. 

That way the values that get set into the Legend can be fixed as averaged values, and the user can modify other properties.

Reply:
I'll just delete the legend_parameters and let the legend property handle everything as discussed.